components/style/build_gecko.rs

```diff
@@ -517,6 +517,49 @@ mod bindings {
         write_binding_file(builder, structs_file(build_type), &fixups);
     }
 
+    pub fn setup_logging() {
+        use log;
+        use std::fs;
+
+        struct BuildLogger {
+            file: Option<Mutex<fs::File>>,
+            filter: String,
+        }
+
+        impl log::Log for BuildLogger {
+            fn enabled(&self, meta: &log::LogMetadata) -> bool {
+                self.file.is_some() && meta.target().contains(&self.filter)
+            }
+
+            fn log(&self, record: &log::LogRecord) {
+                if !self.enabled(record.metadata()) {
+                    return;
+                }
+
+                let mut file = self.file.as_ref().unwrap().lock().unwrap();
+                let _ =
+                    writeln!(file, "{} - {} - {} @ {}:{}",
+                             record.level(),
+                             record.target(),
+                             record.args(),
+                             record.location().file(),
+                             record.location().line());
+            }
+        }
+
+        log::set_logger(|log_level| {
+            log_level.set(log::LogLevelFilter::Debug);
+            Box::new(BuildLogger {
+                file: env::var("STYLO_BUILD_LOG").ok().and_then(|path| {
+                    fs::File::create(path).ok().map(Mutex::new)
```

Comment:
In particular, the ideal would be to find the user's PWD, but that's not exposed as far as I can tell. The workspace root in Gecko isn't also anywhere you'd specially expect your logs to appear, and I guess I can always give another env var to the build script, but seemed unnecessary.